src/main/kotlin/com/humlib/controller/SearchHumansController.kt

```diff
@@ -0,0 +1,61 @@
+package com.humlib.controller
+
+import com.humlib.model.HumanProfile
+import com.humlib.model.HumanProfileTags
+import com.humlib.repository.HumanProfileRepository
+import com.humlib.security.annotations.IsKid
+import org.springframework.data.domain.PageRequest
+import org.springframework.data.domain.Pageable
+import org.springframework.web.bind.annotation.*
+
+@RestController
+@RequestMapping("/search")
+@IsKid
+class SearchHumansController(
+    val humanProfileRepository: HumanProfileRepository
+) {
+
+    @GetMapping
+    fun getAll(): List<HumanProfile> {
+        return humanProfileRepository.findAll().toList()
+    }
+
+    @GetMapping("/contains_all")
+    fun containsAll(
+        @RequestParam pageNo: Int,
+        @RequestParam pageSize: Int,
+        @RequestBody humanProfileTags: HumanProfileTags
+    ): List<HumanProfile> {
```

Comment:
Dieser und der nächset Block sind ja, bis auf die Anzahl der zu matchenden tags, doppelt. Könnte man ja in eine Methode (im Service) auslagern.